openedx/core/djangoapps/course_roles/docs/decisions/0001-course_level_roles.rst

```diff
@@ -0,0 +1,89 @@
+1. Course Level Roles
+######################
+
+
+Status
+******
+
+**Provisional** *2023-09-13*
+
+The status will be updated to *Accepted* upong completion of reimplementation. Related work:
+
+- `Course Roles Setup <https://2u-internal.atlassian.net/browse/ROLES-2>`
+
+
+Context
+*******
+
+There is currently no single functionality being used by the majority of the codebases that are a part of Open edX
+that allows for adding roles (permission sets) for a user or group of users at the course and organization level
+that provides the flexibility to allow users to edit or create new roles (permission sets) that can be assigned to users or groups of users.
+
+
+Decision
+********
+
+Modified LMS table and solution - within LMS/CMS
+------------------------------------------------
+- We will add new tables to manage roles, permission sets, and groups of users that are assigned to roles.
+- We will add additional tables to manage role permissions (to accommodate custom roles).
```

Comment:
Since we are not implementing custom roles initially, can we change this to 
"We will add additional tables, if necessary, to manage custom roles"